hooks/hookexecution/executor_test.go

```diff
@@ -1164,12 +1231,68 @@ func TestExecuteBidderRequestStage(t *testing.T) {
 				},
 			},
 		},
+		{
+			description:           "Group payload not changes with transmitUFPD activity restriction",
```

Comment:
Why isn't this test modifying the payload? This involves the bidder request hook, whose payload has activity support, and you've set it up so the activity infrastructure says transmitting FPD is not allowed. Shouldn't the payload be modified?

Is the issue the fact that this test has always compared `givenBidderRequest` with `expectedBidderRequest` expecting them to be equal, which should still be true even though we are cloning the the bidder request portion of the payload and modifying it?

Reply:
I find this test to be a little redundant and confusing, particularly because it has the exact same inputs and outputs as other test cases except for the fact that there is an activities object input.

What do you think about deleting this test case and instead adding this code block to each of the six stage tests in this file right after the call to `SetAccount`:
```
privacyConfig := getTransmitUFPDActivityConfig("foo", false)
ac := privacy.NewActivityControl(privacyConfig)
exec.SetActivityControl(ac)
```

In this case, we will be taking advantage of the test cases that already exist throughout the file, exercising the activity infrastructure logic for each stage (not a necessity IMO given the nice unit test coverage you have in `execution_test.go` but comes for free here), and introducing some symmetry which I think makes it easier for us to maintain this as it makes the coverage intent easier to discover. On top of that, what you're trying to test here will be exercised for each test case because the `TestApplyHookMutationsBuilder` used in each of the six stage tests already defines a `foo` hook in each stage plan.
Thoughts?

Reply:
Agree, modified!